sklearn/decomposition/_dict_learning.py

```diff
@@ -221,6 +222,26 @@ def _sparse_encode(
     return new_code
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
+        "dictionary": ["array-like"],
+        "gram": ["array-like", None],
+        "cov": ["array-like", None],
```

Comment:
I believe both these need to be `ndarray` objects rather than general `array-like`s.
```suggestion
        "gram": [np.ndarray, None],
        "cov": [np.ndarray, None],
```